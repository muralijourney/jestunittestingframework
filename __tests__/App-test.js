/**
 * @format
 */

import 'react-native';
import React from 'react';
import {fireEvent, render, waitFor} from '@testing-library/react-native';
import App from '../App';
import { Alert } from 'react-native';


// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  renderer.create(<App/>);
});

it('Component: toMatchSnapshot', () => {
  const tree = renderer.create(<App/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Login Flow Check', async ()  => {
  const push = jest.fn();
  jest.spyOn(Alert, 'alert');
  const {getByTestId,getByPlaceholderText} = render(<App/>)
  fireEvent.changeText(getByPlaceholderText(/Email./i), "m@gmail.com")
  fireEvent.changeText(getByPlaceholderText(/Password./i), "1234")
  fireEvent.press(getByTestId('touchable'))
  await waitFor(() => {
    expect(Alert.alert).toHaveBeenCalledWith('Login Success')
  });
});
