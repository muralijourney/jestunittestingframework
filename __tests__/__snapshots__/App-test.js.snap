// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: toMatchSnapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="welcomeScreen"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFC0CB",
        "borderRadius": 30,
        "height": 45,
        "marginBottom": 20,
        "width": "70%",
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Email."
      placeholderTextColor="#003f5c"
      style={
        Object {
          "flex": 1,
          "height": 50,
          "marginLeft": 20,
          "padding": 10,
        }
      }
      testID="#email"
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFC0CB",
        "borderRadius": 30,
        "height": 45,
        "marginBottom": 20,
        "width": "70%",
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Password."
      placeholderTextColor="#003f5c"
      secureTextEntry={true}
      style={
        Object {
          "flex": 1,
          "height": 50,
          "marginLeft": 20,
          "padding": 10,
        }
      }
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "height": 30,
          "marginBottom": 30,
        }
      }
    >
      Forgot Password?
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FF1493",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "marginTop": 40,
        "opacity": 1,
        "width": "80%",
      }
    }
    testID="touchable"
  >
    <Text>
      LOGIN
    </Text>
  </View>
</View>
`;
